pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -288,14 +297,48 @@ func (r *ReconcileClusterPool) Reconcile(request reconcile.Request) (reconcile.R
 	}
 
 	switch drift := reserveSize - int(clp.Spec.Size); {
+	// activity quota exceeded, so no action
+	case clp.Spec.MaxConcurrent != nil && int(*clp.Spec.MaxConcurrent)-len(installingCDs)-numberOfDeletingCDs <= 0:
+		logger.WithFields(log.Fields{
+			"MaxConcurrent": *clp.Spec.MaxConcurrent,
+			"Current":       len(installingCDs) + numberOfDeletingCDs,
+		}).Info("Cannot create/delete clusters as max concurrent quota exceeded.")
 	// If too many, delete some.
 	case drift > 0:
-		if err := r.deleteExcessClusters(installingCDs, readyCDs, drift, logger); err != nil {
+		toDel := drift
+		if clp.Spec.MaxConcurrent != nil {
+			availableActivity := int(*clp.Spec.MaxConcurrent) - len(installingCDs) - numberOfDeletingCDs
+			if toDel > availableActivity {
+				toDel = availableActivity
+			}
+		}
+		if err := r.deleteExcessClusters(installingCDs, readyCDs, toDel, logger); err != nil {
 			return reconcile.Result{}, err
 		}
 	// If too few, create new InstallConfig and ClusterDeployment.
 	case drift < 0:
-		if err := r.addClusters(clp, -drift, logger); err != nil {
+		toAdd := -drift
+		if !hasAvailableCapacity {
+			logger.WithFields(log.Fields{
+				"InUse":    clp.Status.Size,
```

Comment:
InUse seems misleading here, my understanding was Status.Size is the number of clusters currently in the pool, and it would exclude claimed clusters. Suggest CurrentPoolSize?

It might be good to move this to line 278 and include all the details, capacity, poolSize, claimedSize.